### PR TITLE
Refactor integration tests to adopt other catalogs

### PR DESCRIPTION
**Pull request by ajantha-bhat**
`TestContext` was a singleton class coupled with REST catalog. 
This is an attempt to make it more generic to plugin other catalog for test.

### CODE REVIEW

**Comment by bryanck** [comment]
Thanks for doing this. One change I'd like to see is to reuse the shared containers, i.e. MinIO, Kafka, Kafka Connect, so we only have the additional overhead of the extra catalog container per catalog type.

**Comment by ajantha-bhat** [comment]
> One change I'd like to see is to reuse the shared containers, i.e. MinIO, Kafka, Kafka Connect, so we only have the additional overhead of the extra catalog container per catalog type.

kafka connect depends on catalog docker. So, If I have to reuse, then I have to bring up all the catalog containers too before kafka connect? So the `TestContext` becomes generic singleton again. I can't pass catalogType to the singleton classes. 
Hence I went with this approach. Let me know if you find any other clean way of doing this.

**Comment by ajantha-bhat** [comment]
> Thanks for doing this. One change I'd like to see is to reuse the shared containers, i.e. MinIO, Kafka, Kafka Connect, so we only have the additional overhead of the extra catalog container per catalog type.

Done. 
I also thought of using UUID for table names if the tests are executed in parallel. But didn't do as the tests are passing.

File: kafka-connect-runtime/src/test/java/io/tabular/iceberg/connect/TestContextUtil.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.wait.strategy.HttpWaitStrategy;
+import org.testcontainers.utility.DockerImageName;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+
+@SuppressWarnings("rawtypes")
+public final class TestContextUtil {
```
Comment (bryanck) [inline kafka-connect-runtime/src/test/java/io/tabular/iceberg/connect/TestContextUtil.java:52]: I don't feel this change is necessary anymore, you can just add the Nessie catalog container initialization to the original `TestContext`.

File: kafka-connect-runtime/src/test/java/io/tabular/iceberg/connect/rest/RESTIntegrationCdcTest.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.tabular.iceberg.connect.rest;
+
+import io.tabular.iceberg.connect.AbstractIntegrationCdcTest;
+import io.tabular.iceberg.connect.TestConstants;
+
+public class RESTIntegrationCdcTest extends AbstractIntegrationCdcTest {
```
Comment (bryanck) [inline kafka-connect-runtime/src/test/java/io/tabular/iceberg/connect/rest/RESTIntegrationCdcTest.java:24]: I feel we can add a catalog type parameter to the tests so we don't need these subclasses.

Comment (bryanck) [reply kafka-connect-runtime/src/test/java/io/tabular/iceberg/connect/rest/RESTIntegrationCdcTest.java:24]: Another alternative is to combine all of the test classes into one abstract test class so you will only need to create one subclass per catalog. This is similar to Iceberg's [CatalogTest](https://github.com/apache/iceberg/blob/768e516e0b138a6e996fe38fe5be4a0f624862db/core/src/test/java/org/apache/iceberg/catalog/CatalogTests.java#L69). I recently made a change so the tests shouldn't need to be in separate classes anymore.

**Comment by bryanck** [comment]
I think this looks good, thanks again @ajantha-bhat . It might need to be refactored a bit when we submit to the Iceberg but we can deal with that then.

**Review by bryanck — APPROVED** [review]

**Comment by ajantha-bhat** [comment]
> It might need to be refactored a bit when we submit to the Iceberg but we can deal with that then.

I didn't know contributing this project as a module to Iceberg is in the roadmaps. 
nice. Looking forward to it.
This can definitely get better attention if it lives in Iceberg repo.

**Pull request merged**